opensrp-reveal/src/main/java/org/smartregister/reveal/util/RevealMapHelper.java

```diff
@@ -53,16 +86,85 @@ public static void addSymbolLayers(@NonNull Style mMapboxMapStyle, Context conte
         mMapboxMapStyle.addLayer(symbolLayer);
 
-
+        // larval breeding symbol layer
         icon = BitmapFactory.decodeResource(context.getResources(), R.drawable.ic_breeding);
         mMapboxMapStyle.addImage(LARVAL_BREEDING_ICON, icon);
         symbolLayer = new SymbolLayer(LARVAL_BREEDING_LAYER, context.getString(R.string.reveal_datasource_name));
         symbolLayer.setProperties(
                 iconImage(LARVAL_BREEDING_ICON),
                 iconSize(dynamicIconSize));
-
         symbolLayer.setFilter(eq(get(TYPE), StructureType.LARVAL_BREEDING_SITE));
-
         mMapboxMapStyle.addLayer(symbolLayer);
     }
+
+    public static void addCaseIndexBoundary(MapboxMap mapboxMap, Context context) {
+        Style mMapboxMapStyle = mapboxMap.getStyle();
+        try {
+            Expression dynamicIconSize = interpolate(linear(), zoom(),
+                    literal(13.98f), literal(1),
+                    literal(17.79f), literal(3f),
+                    literal(18.8f), literal(4));
+
+            // index case symbol layer
+            if (getInterventionLabel() == R.string.focus_investigation) {
+                Bitmap icon = BitmapFactory.decodeResource(context.getResources(), R.drawable.ic_index_case_target_icon);
+                mMapboxMapStyle.addImage(INDEX_CASE_TARGET_ICON, icon);
+                SymbolLayer symbolLayer = new SymbolLayer(INDEX_CASE_SYMBOL_LAYER, context.getString(R.string.reveal_datasource_name));
+                symbolLayer.setProperties(
+                        iconImage(INDEX_CASE_TARGET_ICON),
+                        iconSize(dynamicIconSize));
+                symbolLayer.setFilter(eq(get(IS_INDEX_CASE), true));
+                if (mMapboxMapStyle.getLayer(INDEX_CASE_SYMBOL_LAYER) == null) {
+                    mMapboxMapStyle.addLayer(symbolLayer);
+                }
+            }
+
+            // index case circle layer
+            GeoJsonSource indexCaseSource = new GeoJsonSource(INDEX_CASE_SOURCE);
+            GeoJsonSource revealSource = mMapboxMapStyle.getSourceAs(context.getString(R.string.reveal_datasource_name));
+            Feature indexCase = getIndexCase(revealSource);
+            if (indexCase != null) {
+                Location centre = (new RevealMappingHelper()).getCenter(indexCase.geometry().toJson());
+                JSONObject feature = new JSONObject(indexCase.toJson());
+                JSONObject geometry = new JSONObject();
+                geometry.put("type", "Point");
+                geometry.put("coordinates", new JSONArray(new Double[]{centre.getLongitude(), centre.getLatitude()}));
+                feature.put("geometry", geometry);
+                indexCaseSource.setGeoJson(Feature.fromJson(feature.toString()));
+
+                CircleLayer circleLayer = new CircleLayer(INDEX_CASE_CIRCLE_LAYER, INDEX_CASE_SOURCE);
+                float circleRadius = calculateZoomLevelRadius(mapboxMap, centre.getLatitude(), 500);
```

Comment:
Remove the 500 and use 1km Radius, also  this should be configurable on server, so implement overriding via server config

Reply:
Done.